presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1451,7 +1445,41 @@ else if (expression instanceof LongLiteral) {
         return analyzedGroupingSets;
     }
 
-    private List<FieldOrExpression> analyzeGroupingColumns(List<Expression> groupingColumns, QuerySpecification node, RelationType tupleDescriptor, AnalysisContext context, List<FieldOrExpression> outputExpressions)
+    private List<Set<Expression>> computeGroupingSetsCrossProduct(List<List<Set<Expression>>> enumeratedGroupingSets, boolean isDistinct)
+    {
+        if (enumeratedGroupingSets.isEmpty()) {
+            return ImmutableList.of();
+        }
+
+        ImmutableList.Builder<Set<Expression>> groupingSetsCrossProduct = ImmutableList.<Set<Expression>>builder()
```

Comment:
Looks like you are not taking advantage of this variable being a `Builder`. I would change this to a list for simplicity.